File: Shift-2D-Grid.cpp
class Solution {
public:
    vector<vector<int>> shiftGrid(vector<vector<int>>& grid, int k) {
        int rows = grid.size();
        int cols = grid[0].size();
        int total = rows * cols;

        vector<vector<int>> result(rows, vector<int>(cols));

        for (int r = 0; r < rows; ++r) {
            for (int c = 0; c < cols; ++c) {
                int newIndex = (r * cols + c + k) % total;
                int newRow = newIndex / cols;
                int newCol = newIndex % cols;
                result[newRow][newCol] = grid[r][c];
            }
        }

        return result;
    }
};

File: Shift-2D-Grid.py
class Solution(object):
    def shiftGrid(self, grid, k):
        """
        :type grid: List[List[int]]
        :type k: int
        :rtype: List[List[int]]
        """
        rows, cols = len(grid), len(grid[0])
        total = rows * cols
        result = [[0] * cols for _ in range(rows)]

        for r in range(rows):
            for c in range(cols):
                new_index = (r * cols + c + k) % total
                new_row = new_index // cols
                new_col = new_index % cols
                result[new_row][new_col] = grid[r][c]

        return result
